lib/mlir/Dialect/Clift/IR/CliftOps.cpp

```diff
@@ -240,179 +241,97 @@ static bool isModuleLevelOperation(Operation *Op) {
 
-      return maybeVisitClassTypeAttr(ContainingOp,
-                                     T.getElementType(),
-                                     RootAttr);
+      return maybeVisitClassTypeAttr(T.getElementType(), RootAttr);
     }
 
     return mlir::success();
   }
 
   template<typename ClassTypeAttr>
-  mlir::LogicalResult visitClassTypeAttr(mlir::Operation *ContainingOp,
-                                         ClassTypeAttr Attr,
-                                         TypeDefinitionAttr RootAttr) {
+  mlir::LogicalResult
+  visitClassTypeAttr(ClassTypeAttr Attr, TypeDefinitionAttr RootAttr) {
     for (FieldAttr Field : Attr.getFields()) {
-      if (visitFieldType(ContainingOp, Field.getType(), RootAttr).failed())
+      if (visitFieldType(Field.getType(), RootAttr).failed())
         return mlir::failure();
     }
     return mlir::success();
   }
 
   // Call visitClassTypeAttr if Attr is a class type attribute.
   // RootAttr is the root class type attribute and is used to detect recursion.
-  mlir::LogicalResult maybeVisitClassTypeAttr(mlir::Operation *ContainingOp,
-                                              TypeDefinitionAttr Attr,
+  mlir::LogicalResult maybeVisitClassTypeAttr(TypeDefinitionAttr Attr,
                                               TypeDefinitionAttr RootAttr) {
     if (auto T = mlir::dyn_cast<StructTypeAttr>(Attr))
-      return visitClassTypeAttr(ContainingOp, T, RootAttr);
+      return visitClassTypeAttr(T, RootAttr);
     if (auto T = mlir::dyn_cast<UnionTypeAttr>(Attr))
-      return visitClassTypeAttr(ContainingOp, T, RootAttr);
+      return visitClassTypeAttr(T, RootAttr);
     return mlir::success();
   }
 
-  mlir::LogicalResult visitTypeAttr(mlir::Operation *ContainingOp,
-                                    TypeDefinitionAttr Attr) {
+  mlir::LogicalResult visitTypeAttr(TypeDefinitionAttr Attr) {
     auto const [Iterator, Inserted] = Definitions.try_emplace(Attr.id(), Attr);
 
     if (not Inserted and Iterator->second != Attr)
-      return ContainingOp->emitError() << "Found two distinct type definitions "
-                                          "with the same ID";
+      return getCurrentOp()->emitError() << "Found two distinct type "
+                                            "definitions with the same ID";
 
-    if (maybeVisitClassTypeAttr(ContainingOp, Attr, Attr).failed())
+    if (maybeVisitClassTypeAttr(Attr, Attr).failed())
       return mlir::failure();
 
     return mlir::success();
   }
 
-  mlir::LogicalResult visitValueType(mlir::Operation *ContainingOp,
-                                     clift::ValueType Type) {
+  mlir::LogicalResult visitValueType(clift::ValueType Type) {
     Type = dealias(Type);
 
     if (not isCompleteType(Type))
-      return ContainingOp->emitError() << "Clift ModuleOp contains an "
-                                          "incomplete type";
+      return getCurrentOp()->emitError() << "Clift ModuleOp contains an "
```

Comment:
Same as above.
`getCurrentOp` or `getCurrentModuleOp`? or is the error message wrong?